### DISCUSSION TITLE
Typing question on session.execute params on sqlalchemy v2

### DISCUSSION THREAD

**Discussion by josumoreno-BP** [sqlalchemy/sqlalchemy#10053] [Usage Questions] [answered]
I have a code similar to this one
```
session = orm.Session(bind=bind)
model_a_ids = session.execute(select(ModelA.id)).mappings().all()
if model_a_ids:
    session.execute(insert(ModelB), model_a_ids) 
```
Mypy is complaining on the last line: `Argument 2 to "execute" of "Session" has incompatible type "Sequence[RowMapping]"; expected "Union[Sequence[Mapping[str, Any]], Mapping[str, Any], None]"  [arg-type]`

I've been checking what a `RowMapping` can be and I saw that the key can be `Union[str, "Column[Any]"]`. I prefer to avoid adding `# type: ignore[arg-type]` but I don't know if there is any other possibility around in this case. Is there?

I also want to ask when the `RowMapping` key can have the form of `Column[Any]`.

Thanks

**Comment by zzzeek** [comment, marked as answer]
I would use either cast() or `[dict(r) for r in result]`

columns can be used as keys in a result set: https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.RowMapping
